modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -251,6 +256,7 @@ interface DOMUtils {
   isEmpty (node: Node, elements?: Record<string, any>): boolean;
   createRng (): Range;
   nodeIndex (node: Node, normalized?: boolean): number;
+  split <T extends Node>(parentElm: Node, splitElm: Node, replacementElm: T, options: SplitOptions): T;
```

Comment:
How does this type for split differ from the type for split below? Could we merge them into one definition (maybe with `options?: SplitOptions`) instead of overloading?

Reply:
This should be the same... so this needs fixing. I believe the correct types for this are:
```
  split <T extends Node>(parentElm: Node, splitElm: Node, replacementElm: T, options?: SplitOptions): T;
  split <T extends Node>(parentElm: Node, splitElm: T, replacementElm?: Node, options?: SplitOptions): T;
```

Reply:
```ts
split <T extends Node>(parentElm: Node, splitElm: T, replacementElm?: undefined, options?: SplitOptions): T
```
for the second one

Reply:
Good point, it should only be passed undefined in that case.